src/Opaleye/Internal/Column.hs

```diff
@@ -10,12 +15,14 @@ import qualified Opaleye.Internal.HaskellDB.PrimQuery as HPQ
 --
 -- Do not use the 'Show' instance of 'Column'.  It will be deprecated
 -- in version 0.6.
-newtype Column pgType = Column HPQ.PrimExpr deriving Show
+data Column_ colType pgType = Column HPQ.PrimExpr deriving Show
+type Column = Column_ NonNullable
 
 -- | Only used within a 'Column', to indicate that it can be @NULL@.
 -- For example, a 'Column' ('Nullable' @PGText@) can be @NULL@ but a
 -- 'Column' @PGText@ cannot.
 data Nullable a = Nullable
+data NonNullable
```

Comment:
Perhaps I misunderstood your outline in #303, but I thought the eventual implementation would be:

```haskell
newtype Column a b = Column HPQ.PrimExpr

data Nullable
data NonNullable
```
Where we then have:  `Column Nullable a` and `Column NonNullable a`

But I cannot switch `Nullable` to kind `*` until we are ready to make the full switch, since it would break everything in the current form.